test/Sign.Core.Test/TestInfrastructure/DirectoryServiceStub.cs

```diff
@@ -0,0 +1,47 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE.txt file in the project root for more information.
+
+namespace Sign.Core.Test
+{
+    internal sealed class DirectoryServiceStub : IDirectoryService
+    {
+        private readonly List<DirectoryInfo> _directories;
+
+        internal IReadOnlyList<DirectoryInfo> Directories { get; }
+
+        internal DirectoryServiceStub()
+        {
+            Directories = _directories = new List<DirectoryInfo>();
+        }
+
+        public DirectoryInfo CreateTemporaryDirectory()
+        {
+            DirectoryInfo directory = new(Path.Combine(Path.GetTempPath(), Path.GetRandomFileName()));
+
+            directory.Create();
```

Comment:
Is there any slim chance of directory with same name already exist by random chance?
Should we have retried logic just in case?

Reply:
There is a chance, but it seems extraordinarily unlikely.  Although adding a retry would make it even less unlikely, just how unlikely is acceptable?  The amount of time to write, test, review, etc. this code doesn't seem worth it given the already low chances of this being a problem.